### PR TITLE
nchat: 3.67 -> 4.13

### PR DESCRIPTION
**Pull request by Aleksanaa**
## Description of changes

Closes https://github.com/NixOS/nixpkgs/pull/276392
Try to fix build on darwin.
Removed an outdated cmake flag.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by sikmir — CHANGES_REQUESTED** [review]
Result of `nixpkgs-review pr 293828` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>nchat</li>
  </ul>
</details>

```
/tmp/nix-build-nchat-4.13.drv-0/source/lib/tgchat/ext/td/td/telegram/Client.h:12:10: fatal error: 'td/telegram/td_api.hpp' file not found
#include "td/telegram/td_api.hpp"
         ^~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

File: pkgs/applications/networking/instant-messengers/nchat/default.nix [hunk 1]
```diff
@@ -30,10 +32,6 @@ stdenv.mkDerivation rec {
     zlib
   ] ++ lib.optionals stdenv.isDarwin [ AppKit Cocoa Foundation ];
 
-  cmakeFlags = [
-    "-DHAS_WHATSAPP=OFF" # go module build required
```
Comment (sikmir) [inline pkgs/applications/networking/instant-messengers/nchat/default.nix:34]: Why? It's still ON by default https://github.com/d99kris/nchat/blob/15b08991bfd8ac10440934691a036bd4a959f4dd/CMakeLists.txt#L46

Comment (Aleksanaa) [reply pkgs/applications/networking/instant-messengers/nchat/default.nix:34]: It is turned off automatically after not finding go https://github.com/d99kris/nchat/blob/15b08991bfd8ac10440934691a036bd4a959f4dd/CMakeLists.txt#L61

**Comment by Aleksanaa** [comment]
Sorry, just didn't want to continue this PR. I'll try again.

**Pull request closed without merging**